integration/project_path_test.go

```diff
@@ -0,0 +1,108 @@
+package integration
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/paketo-buildpacks/occam"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+	. "github.com/paketo-buildpacks/occam/matchers"
+)
+
+func testProjectPath(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect     = NewWithT(t).Expect
+		Eventually = NewWithT(t).Eventually
+
+		pack   occam.Pack
+		docker occam.Docker
+	)
+
+	it.Before(func() {
+		pack = occam.NewPack()
+		docker = occam.NewDocker()
+	})
+
+	context("when built with custom project path set", func() {
+		var (
+			image     occam.Image
+			container occam.Container
+			name      string
+			source    string
+		)
+
+		it.Before(func() {
+			var err error
+			name, err = occam.RandomName()
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		it.After(func() {
+			Expect(docker.Container.Remove.Execute(container.ID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
+			Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
+			Expect(os.RemoveAll(source)).To(Succeed())
+		})
+
+		it("builds and runs correctly", func() {
+			var err error
+
+			source, err = occam.Source(filepath.Join("testdata", "custom_project_path_app"))
+			Expect(err).ToNot(HaveOccurred())
+
+			var logs fmt.Stringer
+			image, logs, err = pack.WithNoColor().Build.
+				WithPullPolicy("never").
+				WithExtensions(
+					settings.Buildpacks.NodeExtension.Online,
+				).
+				WithBuildpacks(
+					settings.Buildpacks.NodeEngine.Online,
+					settings.Buildpacks.BuildPlan.Online,
+				).
+				WithEnv(map[string]string{
+					"BP_NODE_PROJECT_PATH": "hello_world_server",
+				}).
+				WithNetwork("host").
+				WithPullPolicy("always").
+				Execute(name, source)
+			Expect(err).ToNot(HaveOccurred(), logs.String)
+
+			Expect(logs).To(ContainLines(
+				fmt.Sprintf("%s 1.2.3", settings.Extension.Name),
+				"  Resolving Node Engine version",
+				"    Candidate version sources (in priority order):",
+				"      .node-version -> \"16.*\"",
+				"      <unknown>     -> \"\"",
+				"",
```

Comment:
In general I've tried to adopt the pattern of using separate `Expect(logs).To(ContainLines())` statements rather than asserting on `""`.

I've found it to be less brittle, as sometimes extra log lines are inserted which would case the test to fail for unrelated reasons.